amd/src/distractionfree.js

```diff
@@ -0,0 +1,95 @@
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Toggles distraction-free mode.
+ *
+ * @module     block_floatingbutton/distractionfree
+ * @copyright  2025 ISB Bayern
+ * @author     Stefan Hanauska <stefan.hanauska@csg-in.de>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+import * as Storage from 'core/sessionstorage';
+
+/**
+ * Initializes the distraction-free mode toggle button.
+ *
+ * @param {string} toggleButtonId - The ID of the toggle button element.
+ * @param {Array} distractionSelectors - An array of CSS selectors for distractions to hide.
+ * @param {Array} nopaddingSelectors - An array of CSS selectors for elements to remove padding / margin from.
+ * @param {boolean} closedrawers - Whether to close open drawers.
+ */
+export const init = (toggleButtonId, distractionSelectors, nopaddingSelectors, closedrawers) => {
+    let button = document.getElementById(toggleButtonId);
+    button.addEventListener('click', () => {
+        const state = Storage.get('block_floatingbutton/distraction-free-button-state');
+        if (state === 'true') {
```

Comment:
If the "toggle" is not based on the state, but on the fact if there is at least one of the elements hidden by block_floatingbutton, this will also work with enabled `$CFG->cachejs` (of course it would not persist across multiple pages, but it at least works at the current one). Also it would at least work in browsers that for some reasons do not support local storage (comments in the moodle sessionstorage let assume that this rare case exists).
```suggestion
        const someHiddenElement = document.querySelector('.block_floatingbutton-hidden, .block_floatingbutton-nopadding');
        if (someHiddenElement) {
```